### ISSUE TITLE
CIFTI outputs not keeping track of voxels that are not included in brain mask

### ISSUE THREAD

**Issue by jsmentch** [nipreps/fmriprep#2333]
fMRIPrep version 20.2.0, singularity container
dataset: ds001110 sherlock merlin
ran with additional flags:
--task-id MerlinMovie --cifti-output --skull-strip-t1w skip --nthreads 4 --skip_bids_validation --ignore slicetiming

I noticed that after running the cifti outputs through fitlins with a simple contrast, many of the subjects show unusual activations that appear to line up with voxels not included in their brain mask, which varies from subject to subject. see the red regions of sub-19 for an example:
![sub-19_fitlins_output](https://user-images.githubusercontent.com/20958589/99839951-981b7a80-2b39-11eb-898a-1a6d9322014d.png)

If I look at the dtseries.nii cifti file that was the output of fmriprep and plot only the values of the brainstem, it appears that many of the values are close to (but never exactly) zero:
![download (2)](https://user-images.githubusercontent.com/20958589/99840190-137d2c00-2b3a-11eb-9dc7-6d2c6cea9a1b.png)

Is it possible for fmriprep to mark these voxels somehow like as NaNs or make a cifti brainmask label that fitlins can then use, or a simpler way around this? For now I can probably figure out a hacky way around this. Thanks!

**Comment by mgxd** [comment]
Unfortunately, this looks to me like a bigger problem - a failure with either the brain-masking / spatial normalization. Would it be possible to rerun adding the `--output-spaces MNI152NLin6Asym:res-2` option so we can view the intermediate files?

**Comment by jsmentch** [comment]
@mgxd will do!

**Comment by jsmentch** [comment]
I've rerun fmriprep with that option added and here is the report from subject 19:

[sub19_report.zip](https://github.com/nipreps/fmriprep/files/5625964/sub19_report.zip)

Which specific files would be best to share? I can post links to them here or maybe I can move them to somewhere on openmind or elsewhere if that is easier

**Comment by mgxd** [comment]
Ok, taking a look at `sub-19/figures/sub-19_task-MerlinMovie_desc-bbregister_bold.svg`, it seems the BOLD's FOV does not fully capture the brainstem, among other regions, which might explain the zeros you are seeing the CIFTI for that particular region. You can quickly see the regions affected by looking at the carpetplot which leverages the CIFTI when generating them (note the gray horizontal lines)
<img width="1337" alt="Screen Shot 2020-12-02 at 9 17 12 AM" src="https://user-images.githubusercontent.com/14110164/100884176-2cada300-347f-11eb-8755-1e44d1c59042.png">


I'm not entirely sure what's the best way to handle this, as I'm hesitant to change any values within the CIFTI generation. Post-processing the affected regions and changing values near 0 to NaN might be good enough?

**Comment by jsmentch** [comment]
The hacky way around this for me was to set voxels with 0 or close to 0 variance to NaN after fmriprep similar to what you are suggesting. Though I haven't run it through fitlins after that yet.

The points that end up NaNed seem to be what are expected, here they are colored dark red:

<img width="624" alt="Screen Shot 2020-12-02 at 10 25 29 AM" src="https://user-images.githubusercontent.com/20958589/100892988-e2312400-3488-11eb-8722-6039c04a2a1b.png">

**Comment by jsmentch** [comment]
fyi setting them to NaNs doesn't seem to interfere with fitlins, at least it ran alright, so if the values are set to NaN (or zero if that is better?) within fmriprep maybe that is ok as suggested and/or:

I've been looking into this, getting a bit lost, but it might be useful to output for each run a `brain_mask.dlabel.nii` cifti2 file analogous to the `brain_mask.nii.gz`. It's not yet clear to me if there is a more efficient way where this info can be captured in the metadata or if there is a simpler way to apply that already existing brain mask .nii to the cifti file.

If these cifti brain masks are generated then they can be used by users to make a xx% coverage mask or sent into fitlins or other apps if they can accept them as inputs

**Comment by effigies** [comment]
Yes, I think we might have an issue for providing a gifti mask. Should expand it for cifti.
